lib/facility.rb

```diff
@@ -1,14 +1,65 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_accessor :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles, 
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(options = {})
+    @name = options[:name]
+    @address = options[:address]
+    @phone = options[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    while vehicle.registration_date.nil?
+      if vehicle.antique?
+        vehicle.plate_type = :antique
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle?
+        vehicle.plate_type = :ev
+        @collected_fees += 200
+      else
+        vehicle.plate_type = :regular 
+        @collected_fees += 100
+      end
+      vehicle.registration_date = Date.today
+      @registered_vehicles << vehicle
+    end
+  end
```

Comment:
In this method, I think we are missing the check to see if this Facility even offers vehicle registrations before performing it.

Be careful when using a `while` loop as it may lead to an infinite loop. Since the `vehicle` argument that we are passing in will only ever be a single vehicle and not an array or something else that needs to be iterated over in a loop I think a normal `if` statement is the better choice in this scenario.